integration/src/main/java/scripts/testscripts/ResourceLineage.java

```diff
@@ -106,20 +148,167 @@ protected void doUserJourney(
                 // Resource 3 needs different name from resource 2, since they're in same workspace
                 .name(UUID.randomUUID().toString()),
             workspaceId2,
-            resource2Id);
-    UUID resource3Id = resource3CloneResult.getResource().getMetadata().getResourceId();
+            cloneReferencedBqDataset2ResourceId);
+
+    // Assert resource lineage on resource 3.
+    // Now there are two entries. Add second entry for second clone.
+    expectedReferencedBqDatasetLineage.add(
+        new ResourceLineageEntry()
+            .sourceWorkspaceId(workspaceId2)
+            .sourceResourceId(cloneReferencedBqDataset2ResourceId));
+    assertEquals(
+        expectedReferencedBqDatasetLineage,
+        resource3CloneResult.getResource().getMetadata().getResourceLineage());
+  }
+
+  private void cloneControlledBqDataset() throws ApiException, InterruptedException {
+    // Clone controlled bq dataset 1 into second workspace
+    var cloneRequest =
+        new CloneControlledGcpBigQueryDatasetRequest()
+            .name("clonedControlledDataset2")
+            .jobControl(new JobControl().id(UUID.randomUUID().toString()))
+            .destinationDatasetName("cloned_controlled_dataset2")
+            .cloningInstructions(CloningInstructionsEnum.DEFINITION)
+            .destinationWorkspaceId(workspaceId2);
+    CloneControlledGcpBigQueryDatasetResult resource2CloneResult =
+        controlledGcpResourceApi.cloneBigQueryDataset(
+            cloneRequest, getWorkspaceId(), controlledBqDatasetWorkspace1ResourceId);
+    resource2CloneResult =
+        ClientTestUtils.pollWhileRunning(
+            resource2CloneResult,
+            () ->
+                controlledGcpResourceApi.getCloneBigQueryDatasetResult(
+                    getWorkspaceId(), cloneRequest.getJobControl().getId()),
+            CloneControlledGcpBigQueryDatasetResult::getJobReport,
+            Duration.ofSeconds(5));
+    assertJobSuccess(
+        "clone controlled BigQuery dataset 2",
+        resource2CloneResult.getJobReport(),
+        resource2CloneResult.getErrorReport());
+    var cloneControlledBqDataset2ResourceId =
+        resource2CloneResult.getDataset().getDataset().getMetadata().getResourceId();
```

Comment:
nit to make test easier to read/scan. To skim the test, I read the comments at the beginning of each block.

Newline after 189

Before 190

```
// Assert resource lineage on resource 2 in second workspace. There is one entry.
```

Same for cloneControlledGcsBucket